src/libraries/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -152,11 +152,14 @@ public static string ExpandEnvironmentVariables(string name)
 
         public static string GetFolderPath(SpecialFolder folder, SpecialFolderOption option)
         {
-            if (!Enum.IsDefined(folder))
-                throw new ArgumentOutOfRangeException(nameof(folder), folder, SR.Format(SR.Arg_EnumIllegalVal, folder));
+            // We don't need to check if folder is defined here, as GetFolderPathCore will do it.
 
-            if (option != SpecialFolderOption.None && !Enum.IsDefined(option))
-                throw new ArgumentOutOfRangeException(nameof(option), option, SR.Format(SR.Arg_EnumIllegalVal, option));
+            if (option is not SpecialFolderOption.None and not SpecialFolderOption.Create and not SpecialFolderOption.DoNotVerify)
+            {
+                Throw(folder, option);
+                static void Throw(SpecialFolder _, SpecialFolderOption option) =>
```

Comment:
We do not use throw helpers in situations like this

Reply:
> Ok, this can use a small comment. Also, is `SpecialFolder _` really required for this?

No it's unnecessary, it eliminates a `mov`, but in a cold block.